File: elixir/sublist.ex
defmodule Sublist do
  @doc """
  Returns whether the first list is a sublist or a superlist of the second list
  and if not whether it is equal or unequal to the second list.
  """
  def compare(a, b) do
    cond do
      a === b -> :equal
      contain(a, b) -> :superlist
      contain(b, a) -> :sublist
      true -> :unequal
    end
  end

  defp contain(_, []), do: true

  defp contain(a, b) do
    Stream.chunk_every(a, length(b), 1, :discard)
    |> Enum.any?(&(&1 === b))
  end
end

File: clojure/sublist.clj
(ns sublist)

(defn contain
  [list1 list2 ct]
  (cond (< (count list2) ct) false
        (= (take ct list2) list1) true
        :else (contain list1 (rest list2) ct)))


(defn classify
  [list1 list2]
  (cond (= list1 list2) :equal
        (empty? list1) :sublist
        (empty? list2) :superlist
        (contain list1 list2 (count list1)) :sublist
        (contain list2 list1 (count list2)) :superlist
        :else :unequal))
